packages/lesswrong/server/callbacks/userCallbackFunctions.tsx

```diff
@@ -329,19 +329,24 @@ export async function changeDisplayNameRateLimit(validationErrors: CallbackValid
 // slugUpdateBeforeCallbackFunction-Users
 
 /**
- * Handle subscribing/unsubscribing in mailchimp when `subscribedToDigest` is changed, including cases where this
- * happens implicitly due to changing another field
+ * Handle subscribing/unsubscribing in mailchimp when either
+ * `subscribedToDigest` or `subscribedToMailchimp` is changed, including cases
+ * where this happens implicitly due to changing another field
  */
-export async function updateDigestSubscription(data: Partial<DbUser>, {oldDocument, newDocument}: UpdateCallbackProperties<"Users">) {
-  // Handle cases which force you to unsubscribe from the digest:
-  // - When a user explicitly unsubscribes from all emails. If they want they can then explicitly re-subscribe
-  // to the digest while keeping "unsubscribeFromAll" checked
+export async function updateMailchimpSubscription(data: Partial<DbUser>, {oldDocument, newDocument}: UpdateCallbackProperties<"Users">) {
+  // Handle cases which force you to unsubscribe from both:
+  // - When a user explicitly unsubscribes from all emails. If they want they
+  //   can then explicitly re-subscribe while keeping "unsubscribeFromAll"
+  //   checked
   // - When a user deactivates their account
   const unsubscribedFromAll = data.unsubscribeFromAll && !oldDocument.unsubscribeFromAll
   const deactivatedAccount = data.deleted && !oldDocument.deleted
   if (hasDigests && (unsubscribedFromAll || deactivatedAccount)) {
     data.subscribedToDigest = false
   }
+  if (hasNewsletter && (unsubscribedFromAll || deactivatedAccount)) {
+    data.subscribedToNewsletter = false
+  }
```

Comment:
I don't think this should unsubscribe you from the newsletter (I'm pretty on the fence about the digest even, but the newsletter is even more separate from the forum)

Reply:
I agree if they subscribed to the newsletter from somewhere else, but not if they subscribed from the Forum, which they would have done here. I'm not particularly on the fence about the digest, personally.